### PR TITLE
LON-1177

### PR DESCRIPTION
**Pull request by SandraPG85**
Fix drag and drop in tree view with limited collections

### CODE REVIEW

File: VAS.Tests/Core/Common/TestRangeObservableCollection.cs [hunk 1]
```diff
@@ -84,25 +135,27 @@ public void TestInsertRange ()
...
+			Assert.AreEqual (1, notifications);
 		}
 
 		[Test ()]
-		public void TestReplace ()
+		public void TestReset ()
 		{
-			List<int> collectionToReplace = new List<int> { 5, 6, 7 };
-			collection.Replace (collectionToReplace.Select ((arg) => IncrementCounter (arg)).Where (arg => arg >= 0));
+			RangeObservableCollection<int> collectionToReplace = new RangeObservableCollection<int> (new List<int> { 5, 6, 7 });
```
Comment (vmartos) [inline VAS.Tests/Core/Common/TestRangeObservableCollection.cs]: it should work with a list, no need to create a RangeObservableCollection to Reset the list

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs [hunk 2]
```diff
@@ -109,14 +145,14 @@ public void TestAddWithLimitationEnabledWithSort ()
...
 
-			col.Limitation.Model = ll;
 			col.Model.AddRange (players);
 			List<Utils.PlayerDummy> newPlayers = new List<Utils.PlayerDummy> ();
 			newPlayers.Add (new Utils.PlayerDummy { CreationDate = DateTime.Now.AddMinutes (1), Name = "P4" });
 			players.AddRange (newPlayers);
-
 			col.Model.AddRange (newPlayers);
 
+			col.Limitation.Model = ll;
```
Comment (vmartos) [inline VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: why is moved here? maybe change the test method name to something more specific?

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs [hunk 3]
```diff
@@ -125,12 +161,14 @@ public void TestModifyMaxLimitationEnabledWithoutSort ()
...
 			CountLicenseLimitation ll = new CountLicenseLimitation { Enabled = true, Maximum = 3 };
 			col.SortByCreationDateDesc = false;
-			col.Limitation.Model = ll;
 			col.Model.AddRange (players);
+
 			List<Utils.PlayerDummy> newPlayers = new List<Utils.PlayerDummy> ();
 			newPlayers.Add (new Utils.PlayerDummy { CreationDate = DateTime.Now.AddMinutes (1), Name = "P4" });
 			players.AddRange (newPlayers);
 			col.Model.AddRange (newPlayers);
+			col.Limitation.Model = ll;
```
Comment (vmartos) [inline VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: the same why the placement of the col.limitation.Model alters the Test results, maybe u can add the model before and just change the Maximum parameter as the test says, Test**ModifyMaxLimitation**EnabledWithoutSort

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs [hunk 4]
```diff
@@ -142,13 +180,14 @@ public void TestModifyMaxLimitationEnabledWithoutSort ()
...
 			CountLicenseLimitation ll = new CountLicenseLimitation { Enabled = true, Maximum = 3 };
-
-			col.Limitation.Model = ll;
 			col.Model.AddRange (players);
+
 			List<Utils.PlayerDummy> newPlayers = new List<Utils.PlayerDummy> ();
 			newPlayers.Add (new Utils.PlayerDummy { CreationDate = DateTime.Now.AddMinutes (1), Name = "P4" });
 			players.AddRange (newPlayers);
 			col.Model.AddRange (newPlayers);
+			col.Limitation.Model = ll;
```
Comment (vmartos) [inline VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: same as before

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs [hunk 5]
```diff
@@ -161,13 +200,15 @@ public void TestModifyEnabledLimitationEnabledWithoutSort ()
...
 			col.SortByCreationDateDesc = false;
-			col.Limitation.Model = ll;
+
 			col.Model.AddRange (players);
 			List<Utils.PlayerDummy> newPlayers = new List<Utils.PlayerDummy> ();
 			newPlayers.Add (new Utils.PlayerDummy { CreationDate = DateTime.Now.AddMinutes (1), Name = "P4" });
 			players.AddRange (newPlayers);
 			col.Model.AddRange (newPlayers);
-			CheckPlayersInLimitedCollectionBy (new int [] { 1, 2, 0, 3 }, players, col);
+			col.Limitation.Model = ll;
```
Comment (vmartos) [inline VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: the same, the tests says modify Enabled limitation, So you need to set the model first, and then change the model to Enabled False. to check that limitedCollectionVM changes it's visible viewmodels.

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs [hunk 6]
```diff
@@ -179,13 +220,14 @@ public void TestModifyEnabledLimitationEnabledWithSort ()
...
 			CountLicenseLimitation ll = new CountLicenseLimitation { Enabled = false, Maximum = 3 };
 
-			col.Limitation.Model = ll;
 			col.Model.AddRange (players);
 			List<Utils.PlayerDummy> newPlayers = new List<Utils.PlayerDummy> ();
 			newPlayers.Add (new Utils.PlayerDummy { CreationDate = DateTime.Now.AddMinutes (1), Name = "P4" });
 			players.AddRange (newPlayers);
 			col.Model.AddRange (newPlayers);
-			CheckPlayersInLimitedCollectionBy (new int [] { 3, 0, 2, 1 }, players, col);
+			col.Limitation.Model = ll;
```
Comment (vmartos) [inline VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: same modify Enabled property, do not set the model to trigger the change

File: VAS.UI.Gtk2/UI/Common/PlaylistTreeView.cs [hunk 7]
```diff
@@ -160,6 +162,12 @@ protected override void ShowMenu ()
 
 				ViewModel.MovePlaylistElements (toRemove, toAdd, index);
 				return true;
+			} else if (elementsToAdd.Key is PlaylistCollectionVM){
+				PlaylistVM toMove = (PlaylistVM)elementsToRemove.Values.First ().First ();
+				App.Current.EventsBroker.Publish (new MoveCategoryEvent<PlaylistVM> {
```
Comment (vmartos) [inline VAS.UI.Gtk2/UI/Common/PlaylistTreeView.cs]: Actually Playlist are not category events, maybe you can change the name of MoveCategoryEvent to something more generic, like MoveElementEvent or MoveViewModelEvent, etc.

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs (hunk 2 above)
Comment (SandraPG85) [reply VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: I changed the name of some of them but these to tests to remain the same than before the limitation has to be applied at the end to respect the sorting.

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs (hunk 3 above)
Comment (SandraPG85) [reply VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: test updated

File: VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs (hunk 2 above)
Comment (SandraPG85) [reply VAS.Tests/MVVMC/TestLimitedCollectionViewModel.cs]: Change reverted and limited collection updated to apply sorting when a limitation is enabled

File: VAS.UI.Gtk2/UI/Common/PlaylistTreeView.cs (hunk 7 above)
Comment (SandraPG85) [reply VAS.UI.Gtk2/UI/Common/PlaylistTreeView.cs]: renamed

**Pull request merged**